standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/tools/MetaToolObjectStore.java

```diff
@@ -0,0 +1,841 @@
+         "join \"PARTITION_PARAMS\" p on p.\"PART_ID\" = t.\"PART_ID\" where \"PARAM_KEY\" " +
+          "in ('" + StatsSetupConst.TOTAL_SIZE + "', '" + StatsSetupConst.NUM_FILES + "', '" + StatsSetupConst.ROW_COUNT + "') and t.\"TBL_ID\" in (";
+      @Override
+      public List<Void> run(List<Long> input) throws Exception {
+        feedBasicStats(QUERY_TEXT0, input, partedTabs, " group by \"TBL_ID\", \"PARAM_KEY\"");
+        return Collections.emptyList();
+      }
+    });
+  }
+
+  private void feedBasicStats(String queryText0, List<Long> input, Map<Long, MetadataTableSummary> summaryMap, String subQ) {
+    int size = input.size();
+    String queryText = queryText0 + (size == 0 ? "" : repeat(",?", size).substring(1)) + ")" + subQ;
+    if (dbType.isMYSQL()) {
+      queryText = queryText.replace("\"", "");
+    }
+    Object[] params = new Object[size];
+    for (int i = 0; i < input.size(); ++i) {
+      params[i] = input.get(i);
+    }
+    Query<?> query = null;
+    try {
+      query = pm.newQuery("javax.jdo.query.SQL", queryText);
+      List<Object[]> result = (List<Object[]>) query.executeWithArray(params);
+      if (result != null) {
+        for (Object[] fields : result) {
+          Long tabId = Long.parseLong(String.valueOf(fields[0]));
+          MetadataTableSummary summary = summaryMap.get(tabId);
+          feedBasicStats(summary, String.valueOf(fields[1]), fields[2]);
+        }
+      }
+    } finally {
+      if (query != null) {
+        query.closeAll();
+      }
+    }
+  }
+
+  private void feedBasicStats(MetadataTableSummary summary, String key, Object value) {
+    if (summary == null || value == null ||
+        !org.apache.commons.lang3.StringUtils.isNumeric(value.toString())) {
+      return;
+    }
+    long val = Long.parseLong(value.toString());
+    switch (key) {
+    case StatsSetupConst.TOTAL_SIZE:
+      summary.setTotalSize(summary.getTotalSize() + val);
+      break;
+    case StatsSetupConst.ROW_COUNT:
+      summary.setNumRows(summary.getNumRows() + val);
+      break;
+    case StatsSetupConst.NUM_FILES:
+      summary.setNumFiles(summary.getNumFiles() + val);
+      break;
+    default:
+      throw new AssertionError("This should never happen!");
+    }
+  }
+
+  /**
+   * Helper method for getMetadataSummary. Extracting the format of the file from the long string.
+   * @param fileFormat - fileFormat. A long String which indicates the type of the file.
+   * @return String A short String which indicates the type of the file.
+   */
+  private static String extractFileFormat(String fileFormat) {
+    if (fileFormat == null) {
+      return "NULL";
+    }
+    final String lowerCaseFileFormat = fileFormat.toLowerCase();
+    Set<String> fileTypes = new HashSet<>(Arrays.asList("iceberg", "parquet", "orc", "avro", "json", "hbase", "jdbc",
+        "kudu", "text", "sequence", "opencsv", "lazysimple", "passthrough"));
```

Comment:
We have referenced hard-corded file format names way too many times, should we create a final static variables or enum like this: https://github.com/apache/hive/blob/master/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java#L185